src/Schema/AbstractSchemaManager.php

```diff
@@ -309,7 +309,7 @@ protected function doListTableIndexes($table): array
      *
      * @throws Exception
      */
-    public function tablesExist($names)
+    public function tablesExist($names, bool $lowercase = true): bool
```

Comment:
This is a breaking change.

We specifically allow userland code to create custom schema managers. Your change will break existing code that overrides `tableExists()`. You must not add new parameters or native return types to existing non-final methods.

Reply:
Hi, @derrabus 
```
Re: "native return types to existing non-final methods":
```

[Done: https://github.com/doctrine/dbal/pull/5937/commits/627f7e9a6c01d165ff4f7ff930c048ae4bc6873b](https://github.com/doctrine/dbal/pull/5937/commits/627f7e9a6c01d165ff4f7ff930c048ae4bc6873b)

Thank you

Reply:
```
Re: "Your PR description merely repeats the code you have written,
but it does not explain the motivation behind your patch."
```
https://github.com/doctrine/dbal/blob/3.6.x/src/Schema/AbstractSchemaManager.php -> `tablesExist()` -> Current implementation

allows to use the table name only using lowercase characters - lowercase table names.



### Details. Real world example.
Manually creating thousands of dynamically named tables
Case:
table: "users_**d**" -> already exists
table: "users_**D**" -> is NOT exists -> should be manually created `users_D` - dynamically pre-generated name

Checking existed tables before creating required tables:
```
...
$tables = ["user_d", "user_D"];

$connection = $this->getEntityManager()->getConnection();

$schemaManager = $connection->getSchemaManager();

for ($i = 0; $i < count($tables); $i++) {

    $tableName = $tableNames[$i];

    if ($schemaManager->tablesExist([$tableName]) !== true) {

         $this->createAllDbTables->createTable($arguments, $tableName);
         $resultSet = $stmt->executeQuery();
         $result    = $resultSet->fetchAllAssociative();
         ...
    }
    ...
}
...

...
```

For tables:
```
$tableName = "user_d"; // "d" -> lowcased

$tableName = "user_D"; // "D" -> UPPERCASED
```
When called:
```
$schemaManager->tablesExist([$tableName])
```
`true` will be in both cases, although the "users_**D**" table does not yet exist.

Patch - allowing to return the correct result for tables with a name in different case:
```
UPPERCASETABLENAME
CamelCaseTableName
lowercasetablename
Titlecasetablename
```
Thank you